python/grass/temporal/temporal_granularity.py

```diff
@@ -344,164 +423,83 @@ def compute_absolute_time_granularity(maps):
 
     """
 
-    has_seconds = False
-    has_minutes = False
-    has_hours = False
-    has_days = False
-    has_months = False
-    has_years = False
-
-    use_seconds = False
-    use_minutes = False
-    use_hours = False
-    use_days = False
-    use_months = False
-    use_years = False
-
-    delta = []
-    datetime_delta = []
+    # Create a granularity dict with time units of increasing length
+    # that covers all possible keys in the result of compute_datetime_delta
+    # The order of the keys is important so that loops over the dictionalry
+    # can be aborted as soon as a non-zero value is encountered
+    granularity_units = {
+        "second": set(),
+        "minute": set(),
+        "hour": set(),
+        "day": set(),
+        "max_days": set(),
```

Comment:
What does max days represent? some tests fail with max days as a unit. Before, it wasn't a possible option.

If the goal is to make them synonyms for computign the gcd, it could be combined into days (and other conversions needed) before calling gcd at the end.

Reply:
`max_days` was tricky..,

I don`t think I fully understand the purpose of that "unit", but it is now handled properly, I guess...